docs/release.md

```diff
@@ -0,0 +1,78 @@
+# Release Process
+
+This document outlines the release process of the load-test tool.
+
+## Requirements
+
+This release process follows [the release process of the self-hosted version of Mattermost](https://handbook.mattermost.com/operations/research-and-development/product/release-process/feature-release).
+
+The requirements for this release process are the following:
+- The monthly performance comparisons, where we compare the Mattermost version about to be released with the latest stable one, are based on the RC-1 version of the new release.
+- We need a stable tested version of the load-test tool to run the performance comparisons.
+- The RC-1 version of Mattermost is cut at around T-14[^1].
+
+[^1]: Mattermost release process specifies all the release events (feature freeze, RC-1 branch cut, final build cut...) by using the T-X notation, where X is the number of working days before the release day.
+
+As such, we base the timeline of our own release process on the following goal:
+
+> We aim to have a new release ready on T-15.
+
+## Process
+
+The goal of this release process is to have a well-tested version where bugs can be surfaced before we use it for the monthly performance comparisons and other analysis. Then, the main part of it will be the testing process, and we will skip any release whose tests did not pass before the testing period ends.
+
+- T-30 (owned by Dev): Cut release branch.
+- From T-29 to T-16 (owned by QA, assisted by Dev): Testing period.
+    - Run a performance comparison with bounded and unbounded tests, Postgres and MySQL, using the latest stable release of Mattermost as both the base and new builds. The test will pass if and only if:
+        - All four tests finish successfully.
+        - There is no significant difference between the base and new builds (they're the same build, and thus should ideally output the same exact results).
+    - If bugs are found:
+        - Fix them in master.
+        - Cherry-pick them to the release branch.
+        - Run the test again.
+- T-16 end of day (owned by Dev): judge whether the tests successfully passed and decide if the release will happen the next day.
+- T-15 (owned by Dev): If the tests passed, release a new version. If not, skip the release for this month.
```

Comment:
The real performance comparison will still happen, and they should block the Mattermost release if needed. What I meant here is that if we don't happen to have fixed the release *of the load-test tool* at this point (which we have tested with a dummy performance comparison), we will simply use the latest stable version for the real performance comparison, instead of trying to rush the new one in.

Reply:
Yes, but in case of new coverage the two things are sort of coupled. We cannot really use a previous version of the tool to test performance right?

Reply:
Indeed. This is the problem we always face: we need to have coverage of new features, and we need to have a stable tested version of the tool to run the performance comparisons. Feature complete for Mattermost is, according [to the handbook](https://handbook.mattermost.com/operations/research-and-development/product/release-process/feature-release), "approximately 1 month prior to the release day", so T-30. If that rule is followed, and if devs add the corresponding coverage to the load-test tool *at the same time* as the feature is being developed, then we should be good.
That's a lot of *if*s, though. We could try to shorten the testing/debugging/fixing period defined here, maybe leave it in one week, so we have an additional week for finishing the coverage after feature complete in Mattermost.
Thoughts?

Reply:
I think we shouldn't be able to merge any new complex functionality in monorepo until performance tests have been conducted on the PR which means coverage becomes a requirement. I think that should be the standard process, not the opposite (merging and hoping coverage will get in time). It will inevitably mean that some features will get delayed due to this and that's in the nature of things. What we cannot do going forward (in my opinion) is shipping untested software.

Reply:
I agree. This is something that every dev should be aware of. And while we've been seeing more and more coverage added to the tool by other teams, I'm sure we're still missing PRs that may benefit from this.
But let's assume, so we don't increase the scope of this discussion greatly, that this is already happening. For the release process being discussing here, that would mean that we *need* to release a new version (if there are indeed changes) and may end up delaying the product release if we have some unfixed bugs by T-15. I think the step in the release would be something like "communicate to the release manager that the release process will be delayed". I guess at this point we would enter fire-extinguishing mode and try to get the release as soon as possible so that we can run the comparison.
An alternative of course is to be more strict in our own PRs, and thoroughly test all of them (accepting the delay that will cause).
Thoughts?

Reply:
Yes, I think that's reasonable. Also if there's a specific regression in the tool itself unrelated to newly added coverage we could always consider reverting to unblock the release/testing process. I think it may depend on case by case basis but overall I am supportive of your proposal.

Reply:
> if there's a specific regression in the tool itself unrelated to newly added coverage we could always consider reverting to unblock the release/testing process

Yeah, that was part of the "fire-extinguishing mode" in my mind :) Ok, I'll reword it tomorrow and ask for a review again. Thanks for the discussion!